src/Compilers/CSharp/Test/Symbol/Symbols/InterfaceOverriddenOrHiddenMembersTests.cs

```diff
@@ -819,5 +819,361 @@ public interface Test { }
+        {
+            var code = @"
+interface A
+{
+    ref readonly int Property { get ; }
+}
+interface B : A
+{
+    new ref readonly int Property { get; }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics();
+
+            var aClass = comp.GlobalNamespace.GetMember<NamedTypeSymbol>("A");
+            var bClass = comp.GlobalNamespace.GetMember<NamedTypeSymbol>("B");
+
+            var aProperty = aClass.GetMember<PropertySymbol>("Property");
+            var bProperty = bClass.GetMember<PropertySymbol>("Property");
+
+            Assert.Empty(aProperty.OverriddenOrHiddenMembers.OverriddenMembers);
+            Assert.Empty(aProperty.OverriddenOrHiddenMembers.HiddenMembers);
+
+            Assert.Empty(bProperty.OverriddenOrHiddenMembers.OverriddenMembers);
+            Assert.Equal(aProperty, bProperty.OverriddenOrHiddenMembers.HiddenMembers.Single());
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void ImplementingMethodWithRefReadOnlyParameterWillNotProduceAnError()
+        {
+            var code = @"
+interface A
+{
+    void M(ref readonly int x);
+}
+class B : A
+{
+    public void M(ref readonly int x) { }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void ImplementingMethodWithRefReadOnlyReturnTypeWillNotProduceAnError()
+        {
+            var code = @"
+interface A
+{
+    ref readonly int M();
+}
+class B : A
+{
+    protected int x = 0;
+    public ref readonly int M() { return ref x; }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void ImplementingPropertyWithRefReadOnlyReturnTypeWillNotProduceAnError()
+        {
+            var code = @"
+interface A
+{
+    ref readonly int Property { get; }
+}
+class B : A
+{
+    protected int x = 0;
+    public ref readonly int Property { get { return ref x; } }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void ImplementingMethodWithDifferentParameterRefnessWillProduceAnError()
+        {
+            var code = @"
+interface A
+{
+    void M(ref readonly int x);
+}
+class B : A
+{
+    public void M(ref int x) { }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics(
+                // (6,11): error CS0535: 'B' does not implement interface member 'A.M(ref readonly int)'
+                // class B : A
+                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "A").WithArguments("B", "A.M(ref readonly int)").WithLocation(6, 11));
+        }
```

Comment:
Yes.

---
In reply to: [104743796](https://github.com/dotnet/roslyn/pull/17547#discussion_r104743796) [](ancestors = 104743796)